test/unit/component.spec.js

```diff
@@ -0,0 +1,226 @@
+describe('Marionette.Toolkit.Component', function () {
+  beforeEach(function () {
+    this.myRegion = new Backbone.Marionette.Region({
+      el:'body'
+    });
+  });
+
+  // SHOWING A VIEW
+  describe('when showing a component', function () {
+    beforeEach(function () {
+      this.showStub = this.sinon.stub();
+      this.MyViewClass = Marionette.ItemView.extend({});
+      this.MyComponent = Marionette.Toolkit.Component.extend({
+        viewOptions: {
+          template: _.template('<div></div>')
+        },
+        ViewClass: this.MyViewClass
+      });
+      this.myComponent = new this.MyComponent();
+      this.myComponent.on('before:show', this.showStub, function() {});
+      this.myComponent.on('show', this.showStub, function() {});
+    });
+
+    describe('in a specified region "showIn()"', function () {
+      it('should show the component', function () {
+        this.myComponent.showIn(this.myRegion);
+        expect(this.myComponent._isShown).to.equal(true);
+      });
+
+      it('should fire the correct before:show and show methods', function () {
```

Comment:
I think maybe we should make two stubs and two expects.. what if show gets called twice, but before:show doesn't get called?  kinda silly, but still..